fe/fe-core/src/main/java/com/starrocks/alter/Alter.java

```diff
@@ -188,6 +198,135 @@ public void processDropMaterializedView(DropMaterializedViewStmt stmt) throws Dd
             db.writeUnlock();
         }
     }
+    public void processAlterMaterializedView(AlterMaterializedViewStatement stmt)
+            throws DdlException, MetaNotFoundException, AnalysisException {
+        // check db
+        final TableName mvName = stmt.getMvName();
+        final String oldMvName = mvName.getTbl();
+        final String newMvName = stmt.getNewMvName();
+        final RefreshSchemeDesc refreshSchemeDesc = stmt.getRefreshSchemeDesc();
+        String dbName = mvName.getDb();
+        Database db = GlobalStateMgr.getCurrentState().getDb(dbName);
+        if (db == null) {
+            ErrorReport.reportDdlException(ErrorCode.ERR_BAD_DB_ERROR, dbName);
+        }
+
+        db.writeLock();
+        try {
+            MaterializedView materializedView = null;
+            final Table table = db.getTable(oldMvName);
+            if (table instanceof MaterializedView) {
+                materializedView = (MaterializedView) table;
+            }
+
+            if (materializedView == null) {
+                throw new MetaNotFoundException("Materialized view " + mvName + " is not find");
+            }
+            // check materialized view state
+            if (materializedView.getState() != OlapTableState.NORMAL) {
+                throw new DdlException("Materialized view [" + materializedView.getName() + "]'s state is not NORMAL. "
+                        + "Do not allow to do ALTER ops");
+            }
+
+            //rename materialized view
+            if (newMvName != null) {
+                processRenameMaterializedView(oldMvName, newMvName, db, materializedView);
+            } else if (refreshSchemeDesc != null) {
+                //change refresh scheme
+                processChangeRefreshScheme(refreshSchemeDesc, materializedView);
+            } else {
+                throw new DdlException("unsupported modification for materialized view");
+            }
+        } finally {
+            db.writeUnlock();
+        }
+    }
+
+    private void processChangeRefreshScheme(RefreshSchemeDesc refreshSchemeDesc, MaterializedView materializedView)
+            throws DdlException {
+        final MaterializedView.MvRefreshScheme refreshScheme = materializedView.getRefreshScheme();
+        final String refreshType = refreshSchemeDesc.getType().name();
+
+        if (refreshSchemeDesc instanceof SyncRefreshSchemeDesc) {
+            throw new DdlException("unsupported change to SYNC refresh type");
+        }
+
+        if (refreshSchemeDesc instanceof AsyncRefreshSchemeDesc) {
```

Comment:
better to use 
if (refreshSchemeDesc instanceof AsyncRefreshSchemeDesc) { xxx }
else if if (refreshSchemeDesc instanceof SyncRefreshSchemeDesc) { xxx }